frontend/src/metabase/dashboard/components/DashCard/DashCard.jsx

```diff
@@ -56,254 +94,301 @@ const propTypes = {
   isNightMode: PropTypes.bool,
 };
 
-class DashCard extends Component {
-  state = {
-    isPreviewingCard: false,
-  };
-
-  async componentDidMount() {
-    const { dashcard, markNewCardSeen } = this.props;
-
-    // HACK: way to scroll to a newly added card
+function DashCard({
+  dashcard,
+  dashcardData,
+  dashboard,
+  slowCards,
+  metadata,
+  parameterValues,
+  gridItemWidth,
+  totalNumGridCols,
+  mode,
+  isEditing = false,
+  isNightMode = false,
+  isFullscreen = false,
+  isMobile = false,
+  isEditingParameter,
+  clickBehaviorSidebarDashcard,
+  headerIcon,
+  onAddSeries,
+  onRemove,
+  navigateToNewCardFromDashboard,
+  markNewCardSeen,
+  showClickBehaviorSidebar,
+  onChangeLocation,
+  onUpdateVisualizationSettings,
+  onReplaceAllVisualizationSettings,
+  dispatch,
```

Comment:
😱  that's a lot of props, but having them listed at least makes them easier to understand.

Reply:
Yeah, that can easily get horrifying. I still believe introducing [visualization contexts](https://www.notion.so/metabase/30-Visualization-Context-c3527ed8b82b44428dac5b3eb281557f) is a way to go